src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs

```diff
@@ -416,7 +416,7 @@ private async ValueTask<IInstalledAndTransitivePackages> GetInstalledAndTransiti
         /// <param name="projectContextInfos">A collection of projects</param>
         /// <param name="cancellationToken">Cancellation Token</param>
         /// <returns>A collection of Global package folder repositories combined with repositories found in packageFolders from PackageReference projects</returns>
-        private async Task<IReadOnlyList<SourceRepository>> GetAllPackageFoldersAsync(
+        public async ValueTask<IReadOnlyList<SourceRepository>> GetAllPackageFoldersAsync(
```

Comment:
Putting a breakpoint on the `await Task.WhenAll` call a few lines down, when opening PM UI, the breakpoint was hit 4 times, and all 4 times `tasks` contained 1 task, whose state was `WaitingForActivation`. In order words, it did not complete synchronously.

From [a (long) blog post about `ValueTask`](https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/)

> unless the performance implications outweigh the usability implications, Task / Task<TResult>are still preferred

and

>  either you expect synchronous completion to be a very common case, or you’re able to effectively pool objects for use with asynchronous completion

So, this doesn't appear to be one of the situations where `ValueTask` provides benefits.

All this to say I think keeping it as `Task<T>` is better.

Reply:
Thanks for the links. Was just blindly copying from the other methods in the service.